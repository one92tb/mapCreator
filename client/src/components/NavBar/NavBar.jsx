import React, { Component } from "react";
import { NavLink as Link } from "react-router-dom";
import styled from "styled-components";

const Panel = styled.div`
  height: 100%;
  background: #4ddbff;
`;
const Header = styled.div`
  display: flex;
  padding-top: 30px;
  margin-bottom: 70px;
  justify-content: center;
`;
const Logo = styled.img`
  margin-right: 5px;
`;

const Title = styled.h1`
  font-size: 26px;
  display: flex;
  align-items: center;
`;
const User = styled.div``;

const LoginImg = styled.img`
  display: flex;
  margin-left: auto;
  margin-right: auto;
`;
const LoginName = styled.span`
  display: block;
  text-align: center;
  margin-top: 10px;
`;

const Nav = styled.ul`
  margin-top: 100px;
  padding-left: 0;
`;

const NavItem = styled.li`
  list-style-type: none;
  margin: 5px 0;
  height: 50px;
  display: flex;
  align-items: center;
  width: 90%
  margin-left: auto;
  margin-right: auto;


  &:hover{
    background: #00b8e6;
  }
`;

const activeClassName = "nav-item-active";

const NavLink = styled(Link).attrs({ activeClassName })`
  color: #000;
  font-size: 20px;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  padding-left: 10px;
  border-radius: 5px;
  color: #000;

  &:hover {
    text-decoration-line: none;
    color: #000;
  }

  &.${activeClassName} {
    background: #00b8e6;
  }
`;

const Icon = styled.img`
  margin-right: 5px;
`;

class NavBar extends Component {
  render() {
    return (
      <Panel>
        <Header>
          <Logo src={"earth.png"} width={50} height={50} />
          <Title className="logoName">mapCreator</Title>
        </Header>
        <User>
          <LoginImg src={"user.png"} width={130} height={130} />
          <LoginName>UserNick</LoginName>
        </User>
        <Nav>
          <NavItem>
            <NavLink to="/" exact={true} activeClassName={activeClassName}>
              <Icon src={"map.png"} width={32} height={32} /> Map
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/createMarker" activeClassName={activeClassName}>
              <Icon src={"gps.png"} width={32} height={32} />Create Marker
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/test1" activeClassName={activeClassName}>
              <Icon src={"graph.png"} width={32} height={32} />Statistic
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/test2" activeClassName={activeClassName}>
              <Icon src={"list.png"} width={32} height={32} />List
            </NavLink>
          </NavItem>
        </Nav>
      </Panel>
    );
  }
}

export default NavBar;
