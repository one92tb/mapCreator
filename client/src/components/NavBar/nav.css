.inputSearchCity{
position: absolute;
right: 15px;
}
.nav{
  position: relative;
  width: 100%;

}
