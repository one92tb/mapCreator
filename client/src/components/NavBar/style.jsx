import styled from "styled-components";
import { NavLink as Link } from "react-router-dom";
const activeClassName = "nav-item-active";

const Panel = styled.div`
  height: 100%;
  background: #4ddbff;

  @media (max-width: 1199px) {
    display: flex;
    justify-content: space-between;
  }
`;

const Header = styled.div`
  display: flex;
  padding-top: 30px;
  margin-bottom: 70px;
  justify-content: center;

  @media (max-width: 1199px) {
    justify-content: flex-start;
    margin-left: 30px;
    margin-bottom: 0;
    padding: 0;
    align-items: center;
  }
`;

const Logo = styled.img`
  margin-right: 5px;
`;

const Title = styled.h1`
  font-size: 26px;
  display: flex;
  align-items: center;

  @media (max-width: 1199px) {
    margin: 0;
  }
`;

const User = styled.div`
  @media (max-width: 1199px) {
    display: none;
  }
`;

const LoginImg = styled.img`
  display: flex;
  margin-left: auto;
  margin-right: auto;
`;
const LoginName = styled.span`
  display: block;
  text-align: center;
  margin-top: 20px;
`;

const Label = styled.label`
  display: none;

  @media (max-width: 1199px) {
    display: flex;
    cursor: pointer;
    font-size: 30px;
    justify-content: flex-end;
    margin: 0;
    align-items: center;
    margin-right: 30px;
  }
`;

const Nav = styled.ul`
  margin-top: 60px;
  padding-left: 0;

  @media (max-width: 1199px) {
    display: none;
    width: 100%;
    text-align: center;
    z-index: 21;
  }
`;

const Input = styled.input`
  display: none;

  @media (max-width: 1199px) {
    &:checked + ${Nav} {
      display: block;
    }
  }
`;

const NavItem = styled.li`
  list-style-type: none;
  margin: 5px 0;
  height: 50px;
  display: flex;
  align-items: center;
  width: 90%
  margin-left: auto;
  margin-right: auto;


  &:hover{
    background: #00b8e6;
  }
`;

const NavLink = styled(Link).attrs({ activeClassName })`
  color: #000;
  font-size: 20px;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  padding-left: 10px;
  border-radius: 5px;
  color: #000;

  &:hover {
    text-decoration-line: none;
    color: #000;
  }

  &.${activeClassName} {
    background: #00b8e6;
  }
`;

const Icon = styled.img`
  margin-right: 5px;
`;

const LogoutBtn = styled.button`
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-top: 40px;
  background: #00b8e6;
  cursor: pointer;
  height: 40px;
  width: 165px;
  padding: 0.375rem 0.75rem;
  border: 1px solid #00b8e6;
  color: #fff;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;

  @media (max-width: 1199px) {
    display: none;
  }
`;

export {
  Label,
  LogoutBtn,
  Panel,
  Header,
  Logo,
  Title,
  User,
  LoginImg,
  LoginName,
  Nav,
  NavItem,
  NavLink,
  Icon,
  Input
};
