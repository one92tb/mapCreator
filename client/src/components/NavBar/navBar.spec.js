import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import { NavBar, Logout } from "./NavBar";
import { Router } from "react-router-dom";
import history from "../../history";
import ReactRouterEnzymeContext from "react-router-enzyme-context";
import LocalStorageMock from "../../../mocks/localStorageMock";

global.localStorage = new LocalStorageMock();

describe("navBar component", () => {
  let wrapper;

  it("passess all props to navBar component", () => {
    const getSelectedMarker = jest.fn();
    const isPanelSelect = jest.fn();
    const logout = jest.fn();
    const redirect = jest.fn();

    wrapper = shallow(
      <NavBar
        getSelectedMarker={getSelectedMarker}
        isPanelSelect={isPanelSelect}
        logout={logout}
        redirect={redirect}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("handle checkBox method - change status", () => {
    const getSelectedMarker = jest.fn();
    const isPanelSelect = jest.fn();
    const logout = jest.fn();
    const redirect = jest.fn();
    wrapper = shallow(
      <NavBar
        getSelectedMarker={getSelectedMarker}
        isPanelSelect={isPanelSelect}
        logout={logout}
        redirect={redirect}
      />
    );

    const toggleInput = wrapper.find("input");

    toggleInput.simulate("change");
    expect(wrapper.instance().state.checked).toBe(true);
    toggleInput.simulate("change");
    expect(wrapper.instance().state.checked).toBe(false);
  });

  it("handle navLink method - click first link", () => {
    const getSelectedMarker = jest.fn();
    const isPanelSelect = jest.fn();
    const logout = jest.fn();
    const redirect = jest.fn();
    const mockRouter = new ReactRouterEnzymeContext();

    wrapper = mount(
      <NavBar
        getSelectedMarker={getSelectedMarker}
        isPanelSelect={isPanelSelect}
        logout={logout}
        redirect={redirect}
      />,
      mockRouter
    );

    const li = wrapper.find("a").first();
    li.simulate("click");
    expect(getSelectedMarker).toHaveBeenCalledTimes(1);
    expect(wrapper.state().checked).toBe(false);
    expect(isPanelSelect).toHaveBeenCalledTimes(0);
  });

  it("handle navLink method - click second link", () => {
    const getSelectedMarker = jest.fn();
    const isPanelSelect = jest.fn();
    const logout = jest.fn();
    const redirect = jest.fn();
    const mockRouter = new ReactRouterEnzymeContext();

    wrapper = mount(
      <NavBar
        getSelectedMarker={getSelectedMarker}
        isPanelSelect={isPanelSelect}
        logout={logout}
        redirect={redirect}
      />,
      mockRouter
    );

    const li = wrapper.find("a").at(2);
    li.simulate("click");
    expect(getSelectedMarker).toHaveBeenCalledTimes(1);
    expect(wrapper.state().checked).toBe(false);
    expect(isPanelSelect).toHaveBeenCalledTimes(1);
  });

  it("handle Logout method", () => {
    const getSelectedMarker = jest.fn();
    const isPanelSelect = jest.fn();
    const logout = jest.fn();
    const redirect = jest.fn();

    wrapper = shallow(
      <NavBar
        getSelectedMarker={getSelectedMarker}
        isPanelSelect={isPanelSelect}
        logout={logout}
        redirect={redirect}
      />
    );

    const logOutBtn = wrapper.find("button");
    console.log(logOutBtn.html());
    logOutBtn.simulate("click");

    expect(logout).toHaveBeenCalledTimes(1);
    expect(redirect).toHaveBeenCalledTimes(1);
    expect(redirect).toHaveBeenCalledWith("/login");
  });
});
