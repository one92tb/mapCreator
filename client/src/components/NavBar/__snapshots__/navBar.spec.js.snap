// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navBar component passess all props to navBar component 1`] = `
<div>
  <div>
    <div>
      <img
        src="earth.png"
      />
      <h1
        className="logoName"
      >
        mapCreator
      </h1>
    </div>
    <label
      htmlFor="toggle"
    >
      â˜°
    </label>
    <input
      id="toggle"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    isChecked={false}
  >
    <div>
      <img
        src="user.png"
      />
      <span>
        UserNick
      </span>
    </div>
    <button
      onClick={[Function]}
    >
      Sign out
    </button>
    <ul>
      <li
        key="0"
      >
        <a
          activeClassName="nav-item-active"
          exact={true}
          onClick={[Function]}
          to="/"
        >
          <img
            src="map.png"
          />
           
          Map
        </a>
      </li>
      <li
        key="1"
      >
        <a
          activeClassName="nav-item-active"
          onClick={[Function]}
          to="/createMarker"
        >
          <img
            src="gps.png"
          />
           
          Create Marker
        </a>
      </li>
      <li
        key="2"
      >
        <a
          activeClassName="nav-item-active"
          onClick={[Function]}
          to="/statistic"
        >
          <img
            src="graph.png"
          />
           
          Statistic
        </a>
      </li>
      <li
        key="3"
      >
        <a
          activeClassName="nav-item-active"
          onClick={[Function]}
          to="/list"
        >
          <img
            src="list.png"
          />
           
          List
        </a>
      </li>
      <li
        key="4"
      >
        <a
          activeClassName="nav-item-active"
          onClick={[Function]}
          to="/users"
        >
          <img
            src="users.png"
          />
           
          Users
        </a>
      </li>
    </ul>
  </div>
</div>
`;
